prime-router/src/main/kotlin/tokens/OktaAuthentication.kt

```diff
@@ -48,40 +51,53 @@ class OktaAuthentication(private val minimumLevel: PrincipalLevel = PrincipalLev
+            logger.error("Wrong Authentication Verifier being used: ${claimVerifier::class} for $host")
+            return AuthenticatedClaimsResult(null, "")
+        }
+        val claims = claimVerifier.checkClaims(accessToken, minimumLevel, organizationName, oktaSender)
+        if (claims == null) {
+            logger.info("Invalid Authorization Header: ${request.httpMethod}:${request.uri.path}")
+            return AuthenticatedClaimsResult(null, invalidClaim)
+        }
+
+        logger.info("Request by ${claims.userName}: ${request.httpMethod}:${request.uri.path}")
+        return AuthenticatedClaimsResult(claims, null)
+    }
+
+    override fun checkAccess(request: HttpRequestMessage<String?>, sender: String): AuthenticationResult {
+        val oktaSender = true
+        val (claims, status) = checkClaims(request, sender, oktaSender)
+        if (claims != null) {
+            return AuthenticationResult(true, null)
+        }
+        return AuthenticationResult(false, status)
+    }
+
     fun checkAccess(
         request: HttpRequestMessage<String?>,
         organizationName: String = "",
         oktaSender: Boolean = false,
         block: (AuthenticatedClaims) -> HttpResponseMessage
     ): HttpResponseMessage {
-        try {
-            val accessToken = getAccessToken(request)
-            if (accessToken == null) {
-                logger.info("Missing Authorization Header: ${request.httpMethod}:${request.uri.path}")
-                return HttpUtilities.unauthorizedResponse(request, missingAuthorizationHeader)
-            }
-            val host = request.uri.toURL().host
-            setRequest(request)
-            val claimVerifier = authenticationVerifier()
-            if (claimVerifier.requiredHosts.isNotEmpty() && !claimVerifier.requiredHosts.contains(host)) {
-                logger.error("Wrong Authentication Verifier being used: ${claimVerifier::class} for $host")
-                return HttpUtilities.unauthorizedResponse(request)
-            }
-            val claims = claimVerifier.checkClaims(accessToken, minimumLevel, organizationName, oktaSender)
-            if (claims == null) {
-                logger.info("Invalid Authorization Header: ${request.httpMethod}:${request.uri.path}")
-                return HttpUtilities.unauthorizedResponse(request, invalidClaim)
-            }
-
-            logger.info("Request by ${claims.userName}: ${request.httpMethod}:${request.uri.path}")
+        val (claims, status) = checkClaims(request, organizationName, oktaSender)
+        if (claims != null) {
             return block(claims)
-        } catch (ex: Exception) {
-            if (ex.message != null)
-                logger.error(ex.message!!, ex)
-            else
-                logger.error(ex)
-            return HttpUtilities.internalErrorResponse(request)
         }
+        return HttpUtilities.unauthorizedResponse(request, status ?: "")
```

Comment:
The string being passed back needs to be json.  use errormsg from Httputils.  Also, is there something better than empty string if there is no status?

Reply:
@whytheplatypus can you address this?  I'm not certain of the scope of changes to auth.